paasta_tools/cli/cmds/secret.py

```diff
@@ -211,9 +220,36 @@ def paasta_secret(args):
             sys.exit(1)
     else:
         service = args.service
+
+    # Check if "decrypt" and "secrets_for_teams" first to avoid vault auth
+    if args.action == "decrypt":
+        secrets_for_teams_only = is_secrets_for_teams_enabled(
+            service, args.yelpsoa_config_root
+        )
+        if secrets_for_teams_only:
```

Comment:
uber mega nit, if you just do
```suggestion
    if args.action == "decrypt" and is_secrets_for_teams_enabled(
        service, args.yelpsoa_config_root
    ):
```
you save 1 level of indentation (and still avoid the file read if `action != "decrypt"`)